cloud/amazon/ec2_vpc_nat_gateway_facts.py

```diff
@@ -0,0 +1,163 @@
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+module: ec2_vpc_nat_gateway_facts
+short_description: Retrieves AWS VPC Managed Nat Gateway details using AWS methods.
+description:
+  - Gets various details related to AWS VPC Managed Nat Gateways
+version_added: "2.1"
+requirements: [ boto3 ]
+options:
+  nat_gateway_ids:
+    description:
+      - Get details of specific nat gateway IDs
+    required: false
+    default: None
+  filters:
+    description:
+      - A dict of filters to apply. Each dict item consists of a filter key and a filter value.
+        See U(http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeNatGateways.html)
+        for possible filters.
+    required: false
+    default: None
+author: Karen Cheng(@Etherdaemon)
+extends_documentation_fragment:
+  - aws
+  - ec2
+'''
+
+EXAMPLES = '''
+# Simple example of listing all nat gateways
+- name: List all managed nat gateways in ap-southeast-2
+  ec2_vpc_nat_gateway_facts:
+    region: ap-southeast-2
+  register: all_ngws
+
+- name: Debugging the result
+  debug:
+    msg: "{{ all_ngws.result }}"
+
+- name: Get details on specific nat gateways
+  ec2_vpc_nat_gateway_facts:
+    nat_gateway_ids:
+      - nat-1234567891234567
+      - nat-7654321987654321
+    region: ap-southeast-2
+  register: specific_ngws
+
+- name: Get all nat gateways with specific filters
+  ec2_vpc_nat_gateway_facts:
+    region: ap-southeast-2
+    filters:
+      state: ['pending']
+  register: pending_ngws
+
+- name: Get nat gateways with specific filter
+  ec2_vpc_nat_gateway_facts:
+    region: ap-southeast-2
+    filters:
+      subnet-id: subnet-12345678
+      state: ['available']
+  register: existing_nat_gateways
+'''
+
+RETURN = '''
+result:
+  description: The result of the describe.
+  returned: success
+  type: list
+'''
+
+try:
+    import botocore
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+import time
+import json
+
+def date_handler(obj):
+    return obj.isoformat() if hasattr(obj, 'isoformat') else obj
+
+
+def get_nat_gateways(client, module, nat_gateway_id=None):
+    params = dict()
+
+    if module.params.get('filters'):
+        params['Filter'] = []
+        for key, value in module.params.get('filters').iteritems():
+            temp_dict = dict()
+            temp_dict['Name'] = key
+            if isinstance(value, basestring):
+                temp_dict['Values'] = [value]
+            else:
+                temp_dict['Values'] = value
+            params['Filter'].append(temp_dict)
+    if module.params.get('nat_gateway_ids'):
+        params['NatGatewayIds'] = module.params.get('nat_gateway_ids')
+
+    try:
+        result = json.loads(json.dumps(client.describe_nat_gateways(**params), default=date_handler))
+    except Exception as e:
+        module.fail_json(msg=str(e.message))
+
+    return result['NatGateways']
```

Comment:
You need to fix up your result to be snake_cased as this is Ansible standard.

See module utils ec2 for helper function called 'camel_dict_to_snake_dict'

Also see new ec2_snapshot_facts for example